packages/rocketchat-lib/server/functions/saveUser.js

```diff
@@ -130,6 +135,10 @@ RocketChat.saveUser = function(userId, userData) {
 			updateUser.$set.name = userData.name;
 		}
 
+		if (userData.role) {
+			updateUser.$set.roles = [ (userData.role || 'user') ];
+		}
+
```

Comment:
IMHO this method should always override the values. We need methods to add and remove roles. @a5his what do you think?